### ISSUE TITLE
Clean up repository branches

### ISSUE THREAD

**Issue by ngbrown** [rabbitmq/rabbitmq-dotnet-client#155]
@michaelklishin, All the bug-id branches in the code repository is cluttering up what is important.  Once a branch is merged into master, the branch tag can be deleted.

I know that this repository was migrated from a Mercurial repository where branches were hard written into the history, but branches in Git are different.  Git branches are more like bookmarks in Mercurial.

Will everyone concur that the bug-id branches that have been merged get deleted?

**Comment by michaelklishin** [comment]
We can clean up branches that were merged more than N weeks ago, for example, if enough people care.

**Comment by pardahlman** [comment]
This is a very good idea! Why not start with removing all the [stale branches in the repo](https://github.com/rabbitmq/rabbitmq-dotnet-client/branches/stale) to begin with? There are a lot of branches that are 1500+ commits behind :)

In my opinion all branches that have been merged to `master` could be removed. Thought on this?

**Comment by michaelklishin** [comment]
@pardahlman it won't affect our development process but I'm not sure that enough people care. 
We have no shortage of things to work on and this is the first time this issue is brought up in about a year that RabbitMQ has moved to GitHub.

When is it problematic that `origin` (or whatever you call our remote) has a lot of branches? You never check out more than 1% of them most of the time.

Please provide specific arguments. "Cluttering" tolerance is a highly personal thing.

**Comment by pardahlman** [comment]
@michaelklishin -- I think there are a few benefits. Especially, it is easier to get a feeling for the activities in the repo. Whenever I clone a repo, I do a `git branch -r` to see what's going on. For a repo that belongs to an organization, I expect to see a handful branches, like

 * [Autofac](https://github.com/autofac/Autofac/branches) that only have `master` and `develop`
 * [AutoMapper](https://github.com/AutoMapper/AutoMapper/branches), [Serilog](https://github.com/serilog/serilog/branches) that has <5 branches with features
 * [MongoDb](https://github.com/mongodb/mongo-csharp-driver/branches) that has branches for multiple upcoming releases

In my opinion, it is a good idea to have a work-flow that prevents the number of branches to increase infinitely. What good does an old branch from 8 years ago do? Any how many branches will we have 8 years from now? Besides, (it might be a small thing, but... :bowtie:) when I checkout a branch I use auto-complete and with 250+ branches I need to write more.

For [RawRabbit](https://github.com/pardahlman/RawRabbit/branches) (my own repo) I currently have the following branches

* `master` (latest release)
* `release/1.8.3`(upcoming release)
* `feature/new-saga` (long running feature development)

Each feature is developed in a feature branch, then merged to the upcoming release branch and the release branch is merged to `master` upon release. That way, `master` reflects the latest available code, the release branch what's coming (and can easily be compared with master) and feature branches only feature specific stuff.

I can help out removing the branches if you let me know what branches you want to keep :+1:

**Comment by michaelklishin** [comment]
OK, fair enough. We can certainly remove all branches with the old `bug*` convention used in the Mercurial days. Then see what'd be the best way to remove branches based on age.

We'd like to keep

 * `master` (most obviously)
 * `stable`
 * `rabbitmq-dotnet-*` branches that are older than 3 months

We can consider removing branches immediately after they are merged but this has to be done for all of our repositories and I need to convince the team first.

**Comment by ngbrown** [comment]
I've removed all of the `bug*` branches and cleaned up the older `rabbitmq-dotnet-` branches that had already been merged.  There's still a few more really old ones that could be cleaned up.  Take a look at https://github.com/rabbitmq/rabbitmq-dotnet-client/branches/stale.

**Comment by michaelklishin** [comment]
Thank you, @ngbrown. I deleted a few more that are older than 6 months.
